plaidml/edsl/edsl.h

```diff
@@ -1695,5 +1695,29 @@ inline Tensor layer(const std::string& op, const TensorVec& operands, const Laye
   return layer(op, operands, {}, fn, loc);
 }
 
+///
+/// Returns a tensor populated with the index value of the shape and axis specified.
+/// \param dims std::vector<TensorDim>
+/// \param axis size_t
+/// \return Tensor
+///
+
+using loopSingefunc = std::function<Tensor()>;
+using loopMultifunc = std::function<TensorVec()>;
+
+inline Tensor loop(int64_t loopCycle, const TensorVec& operands) {
```

Comment:
I think we need something like lower bound, upper bound and step, and probably a loop iterator.

Reply:
ok, it means we need open all scf.for op's operands in this intrinsic op, right?